api/models.py

```diff
@@ -69,3 +69,13 @@ def convert_values_to_string(cls, elements):
         for key, element in elements.items():
             converted_elements[key] = json.dumps(element)
         return converted_elements
+
+
+@receiver(models.signals.post_save, sender=FormData)
+def render_form_data_html(sender, instance, created, raw, **kwargs):
+    """Renders FormData.html after a new FormData has been created."""
+    if created and not raw:
+        from .serializers import FormDataSerializer
+        data = FormDataSerializer(instance).data
+        data['elements'] = FormData.convert_to_dict(instance.elements)
+        instance.html = HopperForm(data=data).render_as_form()
```

Comment:
If you do this post-save, will the html value actualy be written to the database? Have you tried / checked this?